src/main/java/org/logstash/dissect/DynamicMethodCache.java

```diff
@@ -0,0 +1,20 @@
+package org.logstash.dissect;
+
+import org.jruby.RubyClass;
+import org.jruby.internal.runtime.methods.DynamicMethod;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+final class DynamicMethodCache {
+    private static final int DEFAULT_SIZE = 4;
+    private static final Map<String, DynamicMethod> DYNAMIC_METHOD_MAP = new ConcurrentHashMap<>(DEFAULT_SIZE);
+
+    private DynamicMethodCache() {
+    }
+
+    static DynamicMethod get(final RubyClass rubyPlugin, final String key) {
+        return DYNAMIC_METHOD_MAP.computeIfAbsent(key,
+                rubyPlugin::searchMethod);
```

Comment:
This will make the plugin only work on Java8 are we still supporting that plugin on 5.x which can still run 7.
If we don't we might need to bump to a major version or make the code work with previous jvm.

Reply:
Hmmm. Will revisit.